File: backend/OnlineFruitShop.Core/Entities/Fruit.cs
using System.ComponentModel.DataAnnotations;

namespace OnlineFruitShop.Core.Entities
{
    public class Fruit
    {
        public int Id { get; set; }
        [Required, MaxLength(120)]
        public string Name { get; set; } = string.Empty;
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public decimal DiscountPrice { get; set; }
        public decimal Stock { get; set; }
        public string Category { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public string ImageData { get; set; } = string.Empty;
        public bool IsFeatured { get; set; }
        public string Status { get; set; } = "Available";
        public string PriceUnit { get; set; } = "KG";
    }
}

File: frontend/src/app/models/fruit.ts
export interface Fruit {
  id: number;
  name: string;
  description: string;
  price: number;
  discountPrice: number;
  stock: number;
  category: string;
  imageUrl: string;
  imageData?: string;
  isFeatured: boolean;
  status: string;
  priceUnit: string;
}
